File: 프로그래머스/lv1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.cpp
#include <bits/stdc++.h>

using namespace std;

string func(int a, int n){
    string ret(n, ' ');
    for(int i = 0; i < n; i++){
        if(a & (1 << i)) ret[n-1-i] = '#';
    }
    return ret;    
}

vector<string> solution(int n, vector<int> arr1, vector<int> arr2) {
    vector<string> answer;
    for(int i = 0; i < n; i++){
        answer.push_back(func(arr1[i] | arr2[i], n));
    } 
    
    
    
    return answer;
}

File: 프로그래머스/lv1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.java
class Solution {
    String func(int x, int n){
        String ret = "";        
        for(int i = n-1; i >= 0; i--){
            if((x & (1 << i)) != 0) ret += "#";
            else ret += " ";
        }
        return ret;
    }
    
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];
        
        for(int i = 0; i < n; i++){
            answer[i] = func(arr1[i] | arr2[i], n);            
        }  
        
        return answer;
    }    
    
}